src/deluge/model/global_effectable/global_effectable.cpp

```diff
@@ -572,12 +575,16 @@ ActionResult GlobalEffectable::modEncoderActionForNonExistentParam(int32_t offse
 				ledLevel = (64 + current);
 
 				displayLevel = compressor.setSidechain(lshiftAndSaturate<24>(current + 64));
+				unit = " HZ";
 				break;
 			}
 			indicator_leds::setKnobIndicatorLevel(0, ledLevel);
 		}
-		char buffer[5];
+		char buffer[12];
 		intToString(displayLevel, buffer);
+		if (display->haveOLED()) {
+			strncat(buffer, unit, 11);
```

Comment:
should be 3 - strncat appends up to n characters from unit to buffer. As is it could overflow easily if someone makes a unit that's too long

Reply:
Updated to 4 as 3 was too short.